jdaviz/core/template_mixin.py

```diff
@@ -1314,13 +1318,80 @@ def add_results_from_plugin(self, data_item):
             self.app.data_collection.remove(self.app.data_collection[self.label])
 
         self.app.add_data(data_item, self.label)
+        if ref_data is not None:
+            self.link_results_from_plugin(ref_data)
 
         if self.add_to_viewer_selected != 'None':
             # replace the contents in the selected viewer with the results from this plugin
             # TODO: switch to an instance/classname check?
             self.app.add_data_to_viewer(self.viewer.selected_id,
                                         self.label, clear_other_data=replace)
 
+    def link_results_from_plugin(self, ref_data):
+        dc = self.app.data_collection
+        ref_data = dc[ref_data.selected_item['label']]
+        plugin_data = dc[self.label]
+
+        def _get_pixel_coordinate_indices(data):
```

Comment:
The problem with this strategy is that when linking cube and 2D data (e.g. a moment map to a cube), the 2D pixel components look like `[Pixel Axis 0 [y], Pixel Axis 1 [x]]`, while the cube pixel components look like `[Pixel Axis 0 [z], Pixel Axis 1 [y], Pixel Axis 2 [x]]`, in which case we want to link the 2D y to the 3D z, and the 2D x to the 3D y (or vice versa in the current handling of the moment maps, where we transpose the result array).

Reference my comment [here](https://github.com/spacetelescope/jdaviz/issues/1323#issuecomment-1167641866) for more detail.

Reply:
Well that makes no sense, lol. 2D y to 3D z? What's the point of even having x or y in the pixel components?

Reply:
I have no idea why the letter Glue assigns to each axis is the opposite of the order they appear in the list of components. Doesn't make sense to me either 🤷

Reply:
As far as you know, is there any attribute we can use in a glue data object to determine what the `x` and `y` axes should be? Or is the rule you mentioned (2D y to 3D z...) always the case (except for our moment map)?

Reply:
I'm not aware of a way to set the letter axis designations to be more sensible (from our point of view) on Data creation, although there might be one. This might be a question for @astrofrog.